File: backendApp/src/main/java/com/cinemas/dto/response/HomeCinemaResponse.java
package com.cinemas.dto.response;

import com.cinemas.entities.Cinema;
import com.cinemas.entities.CinemaImages;
import com.cinemas.entities.Movie;
import com.cinemas.entities.Showtimes;
import com.cinemas.enums.StatusCinema;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.FetchType;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class HomeCinemaResponse {
    private Integer id;

    private String name;

    private String slug;

    private String address;

    private String phone;

    private String description;

    private String city;

    private String lat;

    private String lng;

    private StatusCinema status;

    private List<CinemaImages> images;

    private List<HomeShowtimeResponse> days;

    List<SelectOptionReponse> cityList;

    List<SelectOptionReponse<?>> cinema;
}

File: fllmbooking_app/lib/data/models/home_cinema_response.dart
import 'package:fllmbooking_app/data/models/home_showtime_response.dart';

class HomeCinemaResponse {
  int? id;
  String? name;
  String? slug;
  String? address;
  String? lat;
  String? lng;
  List<HomeShowtimeResponse>? days;

  HomeCinemaResponse({
    required this.id,
    required this.name,
    required this.slug,
    required this.address,
    required this.lat,
    required this.lng,
    required this.days,
  });

  factory HomeCinemaResponse.fromJson(Map<String, dynamic> json) {
    return HomeCinemaResponse(
      id: json['id'],
      name: json['name'],
      slug: json['slug'],
      address: json['address'],
      lat: json['lat'],
      lng: json['lng'],
      days: json['days'] != null
          ? (json['days'] as List)
              .map((day) => HomeShowtimeResponse.fromJson(day))
              .toList()
          : null,
    );
  }
}
